src/Pulsar.Client/Internal/ConsumerImpl.fs

```diff
@@ -1135,3 +1143,40 @@ type internal ConsumerImpl<'T> (consumerConfig: ConsumerConfiguration<'T>, clien
                     | Error ex -> reraize ex 
                 } |> ValueTask
 
+
+and internal ZeroQueueConsumerImpl<'T> (consumerConfig: ConsumerConfiguration<'T>, clientConfig: PulsarClientConfiguration, connectionPool: ConnectionPool,
+                           partitionIndex: int, startMessageId: MessageId option, lookup: BinaryLookupService,
+                           startMessageRollbackDuration: TimeSpan, createTopicIfDoesNotExist: bool, schema: ISchema<'T>,
+                           schemaProvider: MultiVersionSchemaInfoProvider option,
+                           interceptors: ConsumerInterceptors<'T>, cleanup: ConsumerImpl<'T> -> unit) =
+    
+    inherit ConsumerImpl<'T>(consumerConfig, clientConfig, connectionPool, partitionIndex, startMessageId, lookup,
+                             TimeSpan.Zero, createTopicIfDoesNotExist, schema, schemaProvider, interceptors, cleanup)
+    
+    override this.ReceiveFsharpAsync() =
+        this.Mb.Post(ConsumerMessage.SendFlowPermits 1)
+        base.ReceiveFsharpAsync()
+    
+    interface IConsumer<'T> with
+        override this.ReceiveAsync() =
+            let prefix = sprintf "consumer(%u, %s, %i)" this.ConsumerId consumerConfig.ConsumerName partitionIndex
+
+            task {
+                match! this.ReceiveFsharpAsync() with
+                | Ok msg when msg.MessageId.Type = MessageIdType.Individual ->
```

Comment:
Looks like this check should be better done inside ReceiveFsharpAsync, otherwise MultiMessageConsumer won't receive any error